### PR TITLE
Support not pad batch in mask-rcnn

### PR DESCRIPTION
**Pull request by yghstill**
DataLoader based on https://github.com/PaddlePaddle/Paddle/pull/31481
- P40 single GPU(batch size=4): gt pad batch ips: `2.85 images/s`, gt **not** pad batch ips: `3.20 images/s`
-  P40 single GPU(batch size=1): gt pad batch ips: `3.27 images/s`, gt **not** pad batch ips: `3.53 images/s`

### CODE REVIEW

File: dygraph/ppdet/modeling/backbones/resnet.py [hunk 1]
```diff
@@ -527,6 +528,8 @@ def out_shape(self):
 
     def forward(self, inputs):
         x = inputs['image']
+        if self.training and isinstance(x, np.ndarray):
```
Comment (jerrywgz) [inline dygraph/ppdet/modeling/backbones/resnet.py]: 统一放到architecture里to_tensor？

File: dygraph/ppdet/modeling/proposal_generator/target.py [hunk 2]
```diff
@@ -265,32 +272,42 @@ def generate_mask_target(gt_segms, rois, labels_int32, sampled_gt_inds,
...
         inds_per_im = sampled_gt_inds[k]
         inds_per_im = paddle.gather(inds_per_im, fg_inds)
 
-        gt_segms_per_im = paddle.gather(gt_segms_per_im, inds_per_im)
+        rois_per_im = rois[k]
+        gt_segms_per_im = gt_segms[k]
+        if isinstance(gt_segms_per_im, list):
+            new_segm = []
+            for poly in gt_segms_per_im:
+                tmp_p = [np.array(p_p) for p_p in poly]
```
Comment (jerrywgz) [inline dygraph/ppdet/modeling/proposal_generator/target.py]: 这里的p_p本身就是numpy?

Comment (yghstill) [reply dygraph/ppdet/modeling/proposal_generator/target.py]: done

File: dygraph/ppdet/modeling/proposal_generator/target.py [hunk 3]
```diff
@@ -35,6 +35,8 @@ def rpn_anchor_target(anchors,
     tgt_deltas = []
     for i in range(batch_size):
         gt_bbox = gt_boxes[i]
+        if isinstance(gt_bbox, np.ndarray):
```
Comment (jerrywgz) [inline dygraph/ppdet/modeling/proposal_generator/target.py:38]: gt_bbox is tensor here

File: dygraph/ppdet/modeling/proposal_generator/target.py [hunk 4]
```diff
@@ -140,6 +142,9 @@ def generate_proposal_target(rpn_rois,
     for i, rpn_roi in enumerate(rpn_rois):
         gt_bbox = gt_boxes[i]
         gt_class = gt_classes[i]
+        if isinstance(gt_classes, list):
```
Comment (jerrywgz) [inline dygraph/ppdet/modeling/proposal_generator/target.py:145]: gt_classes is tensor as well?

File: dygraph/ppdet/modeling/proposal_generator/target.py [hunk 5]
```diff
@@ -233,8 +238,10 @@ def polygons_to_mask(polygons, height, width):
 
 def rasterize_polygons_within_box(poly, box, resolution):
     w, h = box[2] - box[0], box[3] - box[1]
-
-    polygons = copy.deepcopy(poly)
+    if isinstance(poly[0], list):
```
Comment (jerrywgz) [inline dygraph/ppdet/modeling/proposal_generator/target.py:241]: 这里逻辑能否统一？

File: dygraph/ppdet/modeling/proposal_generator/target.py [hunk 6]
```diff
@@ -265,36 +272,47 @@ def generate_mask_target(gt_segms, rois, labels_int32, sampled_gt_inds,
...
+            inds_per_im = inds_per_im.numpy()
+            for i in inds_per_im:
+                new_segm.append(gt_segms_per_im[i])
+            fg_inds_new = fg_inds.reshape([-1]).numpy()
+            results = []
+            for j in fg_inds_new:
+                results.append(
+                    rasterize_polygons_within_box(new_segm[j], boxes[j],
+                                                  resolution))
+        else:
```
Comment (jerrywgz) [inline dygraph/ppdet/modeling/proposal_generator/target.py:301]: strip_pad could be removed

File: dygraph/ppdet/modeling/proposal_generator/target_layer.py [hunk 7]
```diff
@@ -41,7 +42,10 @@ def __call__(self, inputs, anchors):
         anchor_box (Tensor): [num_anchors, 4], num_anchors are all anchors in all feature maps.
         """
         gt_boxes = inputs['gt_bbox']
-        batch_size = gt_boxes.shape[0]
+        if isinstance(gt_boxes, list):
```
Comment (jerrywgz) [inline dygraph/ppdet/modeling/proposal_generator/target_layer.py:45]: gt_boxes is list here
